postmarketOS/pmbootstrap
# Fix conflict with mkbootimg and android-tools

### Motivation
Alpine ships `mkbootimg` with the `android-tools` package now. This conflicts with the `mkbootimg` fork from osm0sis (see #441).

### Changes
* Rename `mkbootimg` to `mkbootimg-osm0sis` (aport and binary name)
* `mkbootimg-osm0sis`: provides `mkbootimg` now (so we don't need to change all the device aports), update version to 2018.05.10
* Adjust our `mkinitfs` script to call `mkbootimg-osm0sis`
* Better aport description

### Failing test case can be ignored here

`test_soname_bump` [does not pass](https://travis-ci.org/postmarketOS/pmbootstrap/builds/391009778#L834):
> RuntimeError: One or more packages need to be rebuilt, because a library they link against had an incompatible upgrade (soname bump). Run 'pmbootstrap pkgrel_bump --auto' to automatically increase the pkgrel in order to trigger a rebuild. If this test case failed during a pull request, the issue needs to be fixed on the 'master' branch first, then rebase your PR on 'master' afterwards.


But this is not an issue in this case, here's why:
* Many device packages depend on `mkbootimg`
* We have a binary `mkbootimg` package in the repositores, which satifies this requirement (the old package before this PR)
* `pkgrel_bump --auto` finds missing dependencies by looking at which packages provide dependencies.
* after this commit, `mkbootimg-osm0sis` provides `mkbootimg`, so the code complains that `mkbootimg-osm0sis` does not have a binary package
* it can not recognize that it's fine in this case, because `mkbootimg` has binary packages and provides the dependency as well

I've tested locally that the test case passes as soon as binary packages of `mkbootimg` are built for `armhf` and `aarch64`, which is what I will do right after the merge.

### How to test

* Run `pmbootstrap bootimg_analyze`
* Run `pmbootstrap install` for a device that needs the `bootimg` format (e.g. `lg-mako`)

(Without this PR, this crashes due to the conflict with `android-tools` and the old `mkbootimg`, and with this PR it works again.)



---
[x] Merge on GitHub (see <https://postmarketos.org/merge>)
